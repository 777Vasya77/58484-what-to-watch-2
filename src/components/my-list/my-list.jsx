import React from 'react';
import PropTypes from 'prop-types';
import Logo from '~/components/logo/logo';
import PageFooter from '~/components/page-footer/page-footer';
import UserBlock from '~/components/user-block/user-block';
import {connect} from 'react-redux';
import {selectors} from '~/selectors/selectors';
import {operations} from '~/operations/oparations';
import Loader from '~/components/loader/loader';
import MoviesList from '~/components/movies-list/movies-list';
import withActiveItem from '~/hocs/with-active-item/with-active-item';

const MoviesListWrapped = withActiveItem(MoviesList);

const MyList = (props) => {
  const {films, isLoading} = props;
  return (
    <div className="user-page">
      <header className="page-header user-page__head">
        <Logo />
        <h1 className="page-title user-page__title">My list</h1>

        <UserBlock />
      </header>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>
        {
          isLoading
            ? <Loader />
            : <MoviesListWrapped filmsList={films} />
        }
      </section>

      <PageFooter />
    </div>
  );
};

MyList.propTypes = {
  loadFavoriteFilms: PropTypes.func,
  films: PropTypes.arrayOf(PropTypes.exact({
    id: PropTypes.number,
    name: PropTypes.string,
    posterImage: PropTypes.string,
    previewImage: PropTypes.string,
    backgroundImage: PropTypes.string,
    backgroundColor: PropTypes.string,
    description: PropTypes.string,
    rating: PropTypes.number,
    scoresCount: PropTypes.number,
    director: PropTypes.string,
    starring: PropTypes.arrayOf(PropTypes.string),
    runTime: PropTypes.number,
    genre: PropTypes.string,
    released: PropTypes.number,
    isFavorite: PropTypes.bool,
    videoLink: PropTypes.string,
    previewVideoLink: PropTypes.string,
  })),
  isLoading: PropTypes.bool,
};

const mapStateToProps = (state) => ({
  isLoading: selectors.user.myFilmListIsLoading(state),
  films: selectors.user.myFilmListSelectors(state),
});

const mapDispatchToProps = (dispatch) => ({
  loadFavoriteFilms: () => dispatch(operations.user.loadFavoriteFilms())
});

export {MyList};

export default connect(mapStateToProps, mapDispatchToProps)(MyList);
