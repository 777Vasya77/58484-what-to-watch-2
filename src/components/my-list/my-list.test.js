import React from 'react';
import renderer from 'react-test-renderer';
import {MyList} from '~/components/my-list/my-list';

jest.mock(`~/components/user-block/user-block`, () => `user-block`);
jest.mock(`~/components/movies-list/movies-list`, () => `movies-list`);
jest.mock(`react-router-dom`, () => ({
  Link: `Link`
}));

describe(`MyList component tests`, () => {
  it(`Component render correctly`, () => {
    const tree = renderer
        .create(
            <MyList />
        ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
