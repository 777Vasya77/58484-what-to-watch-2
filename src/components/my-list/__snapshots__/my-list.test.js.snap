// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList component tests Component render correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <Link
        className="logo__link logo__link--light"
        style={
          Object {
            "borderColor": "unset",
            "color": "unset",
          }
        }
        to="/"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </Link>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <user-block />
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <movies-list
      activeItem={null}
      onChangeActiveItem={[Function]}
    />
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <Link
        className="logo__link logo__link--light"
        style={
          Object {
            "borderColor": "unset",
            "color": "unset",
          }
        }
        to="/"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </Link>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
